File: BOJ(Java 공부용)/10828.java
import java.io.*;
import java.util.ArrayDeque;
import java.util.ArrayList;
import java.util.Deque;

public class Main {
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] input = br.readLine().split(" ");
        int n = Integer.parseInt(input[0]);
        Deque<Integer> stack = new ArrayDeque<>();
        for(int i=0;i<n;i++) {
            String[] s = br.readLine().split(" ");
            if(s[0].equals("push")) {
                int num = Integer.parseInt(s[1]);
                stack.add(num);
            } else if(s[0].equals("top")) {
                if(stack.isEmpty()) {
                    bw.write("-1\n");
                    bw.flush();
                } else {
                    bw.write(String.valueOf(stack.peekLast())+"\n");
                    bw.flush();
                }
            } else if(s[0].equals("size")) {
                bw.write(String.valueOf(stack.size()) +"\n");
                bw.flush();
            } else if(s[0].equals("empty")) {
                if(stack.isEmpty()) {
                    bw.write("1\n");
                    bw.flush();
                } else {
                    bw.write("0\n");
                    bw.flush();
                }
            } else if(s[0].equals("pop")) {
                if(stack.isEmpty()) {
                    bw.write("-1\n");
                    bw.flush();
                } else {
                    bw.write(String.valueOf(stack.pollLast())+"\n");
                    bw.flush();
                }
            }
        }
    }
}

File: BOJ(Kotlin)/10828.kt
import java.io.BufferedReader
import java.io.BufferedWriter
import java.io.InputStreamReader
import java.io.OutputStreamWriter
import java.util.*
import kotlin.math.max
import kotlin.math.min

var ans = 0
fun main(args: Array<String>) {
    val br = BufferedReader(InputStreamReader(System.`in`))
    val bw = BufferedWriter(OutputStreamWriter(System.out))

    var st = StringTokenizer(br.readLine())
    val n = st.nextToken().toInt()
    var deque  : Deque<Int> = ArrayDeque<Int>()
    for (i in 1..n) {
       val addList : List<String> = br.readLine().split(" ")
       val command = addList.get(0)
        when (command) {
            "push" -> {
                deque.addFirst(addList.get(1).toInt())
            }
            "top" -> {
                if (deque.size == 0) {
                    println(-1)
                }
                else {
                    println(deque.first)
                }
            }
            "size" -> {
                println(deque.size)}
            "empty"-> {
                if (deque.size == 0 ) {
                    println(1)
                }
                else {
                    println(0)
                }
            }
            "pop" -> {
                if (deque.size == 0) {
                    println(-1)
                }
                else {
                    println(deque.pollFirst())
                }
            }
        }
    }
}